internal/accrual/api/client/client.go

```diff
@@ -0,0 +1,120 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"go.uber.org/zap"
+	"io"
+	"net/http"
+	"strings"
+)
+
+const ServiceAccrualURI = "/api/orders/{number}"
+const ServiceName = "Accrual"
+
+type ErrorNoContent string
+type ErrorTooManyRequests string
+type ErrorInternalServerError string
+type ErrorUndefined struct {
+	code int
+}
+
+func (err ErrorNoContent) Error() string {
+	return fmt.Sprintf("Сервис %s вернул пустой ответ", ServiceName)
+}
+
+func (err ErrorTooManyRequests) Error() string {
+	return fmt.Sprintf("Сервис %s перегружен запросами", ServiceName)
+}
+
+func (err ErrorInternalServerError) Error() string {
+	return fmt.Sprintf("Сервис %s сломался", ServiceName)
+}
+
+func (err ErrorUndefined) Error() string {
+	return fmt.Sprintf("Сервис %s вернул не обработаный код ошибки %d", ServiceName, err.code)
+}
+
+type AccrualClient struct {
+	serviceURL string
+	logger     *zap.SugaredLogger
+	ctx        context.Context
+}
+
+func NewAccrualClient(serviceURL string, logger *zap.SugaredLogger, ctx context.Context) AccrualClientInterface {
```

Comment:
Не стоит возвращать ссылку на интерфейс так как в этом нет необходимости. Так как Go использует  duck typing, реализация будет автоматически реализовывать интерфейс. Вообще стоит подумать о том чтобы определить интерфейс там где он используется (в сервисе или хэндлере). 

```suggestion
func NewAccrualClient(serviceURL string, logger *zap.SugaredLogger, ctx context.Context) *AccrualClient {
```